docs_src/microservices/device/details/AutoEvents.md

```diff
@@ -3,5 +3,70 @@ title: Device Service - Auto Events
 ---
 
 # Device Service - Auto Events
+## Concept
+![image](Auto_Events.png)
+In the Operational Technology (OT) working field, there's a need to check/monitor the status of the actual device/sensor, for example, when the temperature reaches 30°C, the Air Conditioner will turn on itself. However, it is time-consuming to check sensor's status manually so edgex provides this functionality called <code>AutoEvents</code> to check their statuses automatically and periodically.
 
-coming soon
\ No newline at end of file
+From the diagram above, <code>Autoevents</code> is used to define how often events/readings are collected to be sent to core data from the device service. When the device service receives the information of devices being created, updated or deleted from core metadata, it will get reading from the real device automatically and send event/reading to MessageBus when App service/Core Data receive them.
+
+<code>Autoevents</code> is an optional feature when creating a devices, each device may or may not have multiple autoevents associated with it. An AutoEvent has the following fields:
+
+- **sourceName**: the name of a deviceResource or deviceCommand indicating what to read.
+- **interval**: a string indicating the time to wait between reading, expressed as an integer followed by units of ms, s, m or h.
+- **onchange**: a boolean: if set to true, only generate new events if one or more of the contained readings has changed since the last event.
+
+## Example and Usage
+
+The AutoEvent is defined in the `autoEvents` section of the device definition file:
+
+!!! example
+    ```yaml
+    device:
+      name: device-demo
+      adminState: UNLOCKED
+      operatingState: UP
+      serviceName: device-virtual
+      profileName: virtual-profile
+      protocols: virtual
+      autoEvents:
+      - interval: 30s 
+        onChange: false
+        sourceName: temperature
+    ```
+
+After service startup, query core-data's API to query the number of events:
+
+``` bash
+curl http://localhost:59880/api/v3/event/all
+```
+
+The results show that the service auto-executes the command every 30 seconds:
+
```

Comment:
use a clear example to show the auto event result.  one event doesn't show the service auto-executes the command every 30 seconds.

Reply:
updated